CRM/Financial/Form/BatchTransaction.php

```diff
@@ -158,8 +168,8 @@ public function buildQuickForm() {
   }
 
   public function setDefaultValues() {
-    // do not setdefault unless it is open batch
-    if ($this->_batchStatusId != 1) {
+    // do not setdefault unless it is open/reopened batch
+    if (!in_array($this->_batchStatusId, array($this->batchStatuses['Open'], $this->batchStatuses['Reopened']))) {
```

Comment:
best idea would be to check
if (!in_array($this->_batchStatus, array('Open', 'Reopened'))) {